File: Source/JapaneseLanguageTools.Contracts/Models/Requests/SubmitCompletedWordExerciseDetailsRequest.Model.cs
using System.Collections.Generic;

using JapaneseLanguageTools.Contracts.Models.Responses;

namespace JapaneseLanguageTools.Contracts.Models.Requests;

public class SubmitCompletedWordExerciseDetailsRequestModel
{
    public IList<WordExerciseModel> CompletedItems { get; set; } = [];

    public required GenerateWordExerciseBatchResponseModel OriginalResponse { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/SubmitCompletedWordExerciseDetailsRequest.Model.ts
import WordExerciseModel from "@/models/WordExercise.Model";
import GenerateWordExerciseBatchResponseModel from "@/models/responses/GenerateWordExerciseBatchResponse.Model";

export default interface SubmitCompletedWordExerciseDetailsRequestModel {
  completedItems: WordExerciseModel[];

  originalResponse: GenerateWordExerciseBatchResponseModel;
}
